### PR TITLE
Task/81 missing account type

### PR DESCRIPTION
**Pull request by fboucquez**
Hi @rg911 @fullcircle23 

This is the pull request for https://github.com/nemtech/nem2-sdk-java/issues/81 that adds account type to the model object and mappers.

Let me know what do you think

Cheers
Fernando

### CODE REVIEW

File: sdk-core/src/main/java/io/nem/sdk/model/account/AccountType.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    /**
+     * Unlinked.
+     */
+    UNLINKED(0),
+
+    /**
+     * Balance-holding account that is linked to a remote harvester account.
+     */
+    BALANCE_HOLDING_ACCOUNT_LINKED_TO_REMOTE_HARVESTER_ACCOUNT(1),
```
Comment (rg911) [inline sdk-core/src/main/java/io/nem/sdk/model/account/AccountType.java]: use the catapult server types
https://github.com/nemtech/catapult-server/blob/ccfd268729e8eaa9b928b17b29ae7fbdeca887ae/src/catapult/state/AccountState.h#L30

File: sdk-core/src/main/java/io/nem/sdk/model/account/AccountInfo.java [hunk 2]
```diff
@@ -42,11 +43,13 @@ public AccountInfo(
         BigInteger publicKeyHeight,
         BigInteger importance,
         BigInteger importanceHeight,
-        List<Mosaic> mosaics) {
+        List<Mosaic> mosaics,
```
Comment (rg911) [inline sdk-core/src/main/java/io/nem/sdk/model/account/AccountInfo.java]: Not sure we need mosaics here.  Referring to catapult-server.  https://github.com/nemtech/catapult-server/blob/ccfd268729e8eaa9b928b17b29ae7fbdeca887ae/src/catapult/state/AccountState.h#L45

Instead, `AccountInfo` should be replaced by `AccountState` IMO

File: sdk-core/src/main/java/io/nem/sdk/model/account/AccountType.java (hunk 1 above)
Comment (fboucquez) [reply sdk-core/src/main/java/io/nem/sdk/model/account/AccountType.java]: Thanks, I was looking for those names.

I'll use the uppercase version as Java conventions

https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html

File: sdk-core/src/main/java/io/nem/sdk/model/account/AccountInfo.java (hunk 2 above)
Comment (fboucquez) [reply sdk-core/src/main/java/io/nem/sdk/model/account/AccountInfo.java]: The "getAccountInfo" and "getAccountsInfo" repository methods are wrappers around https://nemtech.github.io/endpoints.html#operation/getAccountInfo

In the spec, they call it AccountInfo. AccountInfo may be a "view" of AccountState, but sdks see AccountInfo (Account state is hidden behind the rest server). 

About mosaics, the server response already includes them and we should give that data to the user as may need it. If not, the user would need to do a separate call to get the mosaics when it's not required. Actually, I cannot see any other method to get mosaics of an account in https://nemtech.github.io/endpoints.html#tag/Account-routes

I reckon it should stay as is.

**Pull request merged**